### PR TITLE
GG-18366 Implemented Migration utility.

### PR DESCRIPTION
**Pull request by akuznetsov-gridgain**

### CODE REVIEW

File: modules/web-console/frontend/package.json [hunk 1]
```diff
@@ -75,7 +75,7 @@
     "popper.js": "^1.14.7",
     "resize-observer-polyfill": "1.5.0",
     "roboto-font": "0.1.0",
-    "rxjs": "6.4.0",
+    "rxjs": "6.5.2",
```
Comment (nva) [inline modules/web-console/frontend/package.json]: Why you change frontend deps in this ticket?

Comment (akuznetsov-gridgain) [reply modules/web-console/frontend/package.json]: Reverted.

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrationFromMongo.java [hunk 2]
```diff
@@ -0,0 +1,546 @@
...
+    /** */
+    @Qualifier("migration")
+    private final MigrateActivitiesRepository activitiesRepo;
+
+    /** */
+    protected MongoDatabase mongoDb;
+
+    /** */
+    @Value("${migration.enabled:false}")
+    private boolean enabled;
```
Comment (nva) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrationFromMongo.java]: Why we need this flag? Check mongo url and that database is empty

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrationFromMongo.java [hunk 3]
```diff
@@ -0,0 +1,546 @@
...
+                    log.warn("Space not found [owner=" + mongoAccId + ", email=" + email + "]");
+
+                    continue;
+                }
+
+                log.info(off(2, "Migrating account [_id=" + mongoAccId + ", email=" + email + "]"));
+
+                UUID accId = UUID.randomUUID();
+
+                Account acc = createAccount(accMongo, accId);
```
Comment (nva) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrationFromMongo.java]: Move this logic to `migrateAccount` and remove `migrateAccountEx`

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrationFromMongo.java [hunk 4]
```diff
@@ -0,0 +1,546 @@
...
+        }
+
+        log.info("Migration started...");
+
+        MongoClient mongoClient = MongoClients.create();
+
+        try {
+            mongoDb = mongoClient.getDatabase(mongoDbName);
+
+            migrateAccounts();
```
Comment (nva) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrationFromMongo.java]: Rename method `migrateAccounts` to `migrate`

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrationFromMongo.java [hunk 5]
```diff
@@ -0,0 +1,546 @@
...
+
+            migrateEx();
```
Comment (nva) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrationFromMongo.java]: This method is not needed, override `migrate`

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrationFromMongo.java [hunk 6]
```diff
@@ -0,0 +1,546 @@
...
+    }
+
+    /**
+     * Extension point for migration of additional objects related to account.
+     *
+     * @param accId Account ID.
+     * @param accMongo Mongo document with account.
+     * @param space Mongo document with account space.
+     */
+    protected void migrateAccountEx(Document accMongo, Document space, UUID accId) {
```
Comment (nva) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrationFromMongo.java]: Remove

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/repositories/ActivitiesRepository.java [hunk 7]
```diff
@@ -23,32 +23,34 @@
...
 import org.apache.ignite.transactions.Transaction;
+import org.springframework.context.annotation.Primary;
 import org.springframework.stereotype.Repository;
 
 import static java.time.ZoneOffset.UTC;
 
 /**
  * Repository to work with activities.
  */
+@Primary
```
Comment (nva) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/repositories/ActivitiesRepository.java]: Why we need `Primary`?

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrateActivitiesRepository.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+     */
+    public MigrateActivitiesRepository(Ignite ignite, TransactionManager txMgr) {
+        super(ignite, txMgr);
+    }
+
+    /**
+     * @param activityKey Activity key.
+     * @param activity Activity to save.
+     */
+    public void migrateActivity(ActivityKey activityKey, Activity activity) {
```
Comment (nva) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrateActivitiesRepository.java]: Please create method `save` in `ActivitiesRepository` and remove this class.

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrationFromMongo.java (hunk 2 above)
Comment (akuznetsov-gridgain) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrationFromMongo.java]: fixed

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/repositories/ActivitiesRepository.java (hunk 7 above)
Comment (akuznetsov-gridgain) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/repositories/ActivitiesRepository.java]: fixed

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrateActivitiesRepository.java (hunk 8 above)
Comment (akuznetsov-gridgain) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrateActivitiesRepository.java]: fixed

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrationFromMongo.java (hunk 4 above)
Comment (akuznetsov-gridgain) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrationFromMongo.java]: I don't think it is a good idea, I need to override this method for "Alerts" migration.

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrationFromMongo.java (hunk 3 above)
Comment (akuznetsov-gridgain) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrationFromMongo.java]: I don't think it is a good idea, I need to override this method for "GG plugin" migration.

Comment (akuznetsov-gridgain) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrationFromMongo.java]: Fixed.

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrationFromMongo.java (hunk 4 above)
Comment (akuznetsov-gridgain) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrationFromMongo.java]: Fixed.

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrationFromMongo.java (hunk 5 above)
Comment (akuznetsov-gridgain) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrationFromMongo.java]: Fixed.

**Pull request closed without merging**